### PR TITLE
Add datasource plugins for Access Advisor and IAM

### PR DESCRIPTION
**Pull request by patricksanders**
This PR begins the addition of an abstraction layer for gathering data from disparate sources, starting with Access Advisor (via Aardvark) and IAM. The datasource classes are singletons and share a common interface. Each datasource can be seeded with data (e.g. for a whole account) to reduce network calls.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,4 @@ tox.ini
 test-reports/*
 config.json
 .eggs/
+dynamodb-data/
```
Comment (castrapel) [inline .gitignore:19]: Should we update the readme to tell the user an alternative way to run local dynamo is to use the docker-compose.yml file ?

File: repokid/cli/repokid_cli.py [hunk 2]
```diff
@@ -307,7 +305,7 @@ def show_scheduled_roles(ctx: click.Context, account_number: str) -> None:
 @cli.command()
 @click.argument("account_number")
 @click.option("--role", "-r", required=False, type=str)
-@click.option("--all", "-a", default=False, help="Commit changes")
+@click.option("--all", "-a", default=False)
```
Comment (castrapel) [inline repokid/cli/repokid_cli.py:308]: shouldn't this be `is_flag=True` ?
Should we still have `help=` something?

File: repokid/commands/repo.py [hunk 3]
```diff
@@ -148,7 +150,7 @@ def _rollback_role(
     role_name: str,
     config: RepokidConfig,
     hooks: RepokidHooks,
-    selection: int = 0,
+    selection: int = -1,
     commit: bool = False,
 ) -> List[str]:
     """
```
Comment (castrapel) [inline repokid/commands/repo.py:156]: nit (not your doing): Typo: `roll` -> `role`

File: repokid/commands/repo.py [hunk 4]
```diff
@@ -148,7 +150,7 @@ def _rollback_role(
     role_name: str,
     config: RepokidConfig,
     hooks: RepokidHooks,
-    selection: int = 0,
+    selection: int = -1,
```
Comment (castrapel) [inline repokid/commands/repo.py:153]: Why was this changed to -1, and the condition below changed to selection < 0 ?

File: repokid/commands/role.py [hunk 5]
```diff
@@ -235,22 +236,25 @@ def _display_role(
     for permission in permissions:
         service = permission.split(":")[0]
         action = permission.split(":")[1]
-        repoable = permission in role.repoable_services
+        repoable = (
```
Comment (castrapel) [inline repokid/commands/role.py:239]: nit: Can you add a comment here or somewhere else with some example values for role.repoable_services?

I'm imagining a list of permissions like ["sqs:ListQueue", "sns:ListTopic"] , but I'm unsure. 

When would the extra condition be used? `or permission.split(":")[0] in role.repoable_services`

File: repokid/datasource/iam.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        if not result:
+            raise NotFoundError
+        return result
+
+    def seed(self, account_number: str) -> None:
+        fetched_data = self._fetch(account_number)
+        self._data.update(fetched_data)
+
+
+class ConfigDatasource(DatasourcePlugin[str, IAMEntry], Singleton):
```
Comment (castrapel) [inline repokid/datasource/iam.py:60]: Nit: Do we want some TODOs here?

File: repokid/datasource/plugin.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+from typing import Optional
+from typing import TypeVar
+from typing import ValuesView
+from typing import cast
+
+from repokid.plugin import RepokidPlugin
+from repokid.types import RepokidConfig
+
+logger = logging.getLogger("repokid")
+KT = TypeVar("KT")
```
Comment (castrapel) [inline repokid/datasource/plugin.py:31]: nit: Should KT and VT be generics, and put in a different file, so they can be re-used in the future?

**Review by castrapel — APPROVED** [review]
Approved with minor comments.

File: repokid/commands/repo.py (hunk 4 above)
Comment (patricksanders) [reply repokid/commands/repo.py:153]: This was changed because the selections are zero-indexed, so 0 is a valid selection. The bail-out was previously `if not selection`, which would evaluate to `True` for `selection = 0`. Sneaky sneaky.

File: repokid/commands/role.py (hunk 5 above)
Comment (patricksanders) [reply repokid/commands/role.py:239]: Yeah, I'll add something. This logic is super convoluted and needs refactoring. I carried it over from the previous implementation to ensure consistency until testing is more robust.

File: repokid/cli/repokid_cli.py (hunk 2 above)
Comment (patricksanders) [reply repokid/cli/repokid_cli.py:308]: Nice catch!

File: repokid/commands/role.py (hunk 5 above)
Comment (patricksanders) [reply repokid/commands/role.py:239]: Oops, I thought this comment was somewhere else.

`role.repoable_services` is both repoable actions AND services. The colon split is used to check if the service name is in the list. It's gross, but I haven't gotten around to refactoring it yet.

File: repokid/datasource/plugin.py (hunk 7 above)
Comment (patricksanders) [reply repokid/datasource/plugin.py]: Moved them to `repokid.types`. They're supposed to be `TypeVar`, and then `Generic` is used in the class definition, i.e.
```python
class DatasourcePlugin(RepokidPlugin, Generic[KT, VT]):
    [...]
```

File: repokid/commands/role.py (hunk 5 above)
Comment (patricksanders) [reply repokid/commands/role.py]: Alright, I broke up the logic and added a comment to explain what's going on here
